src/Microsoft.OpenApi/Models/References/BaseOpenApiReferenceHolder.cs

```diff
@@ -63,20 +63,20 @@ protected BaseOpenApiReferenceHolder(string referenceId, OpenApiDocument hostDoc
     /// <inheritdoc/>
     public virtual void SerializeAsV3(IOpenApiWriter writer)
     {
-        if (!writer.GetSettings().ShouldInlineReference(Reference))
+        if (Reference is not null && !writer.GetSettings().ShouldInlineReference(Reference))
```

Comment:
I believe we should add the required keyword to the reference field, as it's already assigned in all constructors anyway and not mutable externally. So it can't be null.
This way we can clean up all those extra validations